### PR TITLE
I1 4 merchant navigation

### PR DESCRIPTION
**Pull request by D-Lessenden**
Need to rename the merchant controller

### CODE REVIEW

File: spec/features/navigation_spec.rb [hunk 1]
```diff
@@ -43,4 +43,67 @@
...
+                          password: '@%)abc123#$.',
+                          role: 1)
+
+      allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(merchant_1)
+
+
+      visit '/items'
+
+      within 'nav' do
+        expect(page).to have_content("Merchant Dashboard")
```
Comment (ryandbarnett) [inline spec/features/navigation_spec.rb]: I think it is more appropriate that have_content be have_link.

Comment (ryandbarnett) [reply spec/features/navigation_spec.rb]: I think this test is missing the "I see the same links as a regular user" part of user story 4.

Comment (ryandbarnett) [reply spec/features/navigation_spec.rb]: - a link to return to the welcome / home page of the application ("/")
- a link to browse all items for sale ("/items")
- a link to see all merchants ("/merchants")
- a link to my shopping cart ("/cart")
- a link to my profile page ("/profile")
- a link to log out ("/logout")

File: spec/features/navigation_spec.rb [hunk 2]
```diff
@@ -43,4 +43,67 @@
...
+
+      within 'nav' do
+        click_link 'Merchant Dashboard'
+        expect(current_path).to eq('/merchant')
+      end
+    end
+  end
+
+
+  describe 'As an Admin' do
```
Comment (ryandbarnett) [inline spec/features/navigation_spec.rb]: This is part of user story 5. 

Looks good but it should also test for "I see the same links as a regular user"

- a link to return to the welcome / home page of the application ("/")
- a link to browse all items for sale ("/items")
- a link to see all merchants ("/merchants")
- a link to my profile page ("/profile")
- a link to log out ("/logout")

**Review by ryandbarnett — CHANGES_REQUESTED** [review]
Please add the I see the same links as a regular user tests to navigation spec. See my comments for more info.

**Review by PhilipDeFraties — APPROVED** [review]
Nice!

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -12,8 +12,16 @@
...
       <%= link_to "All Merchants", "/merchants"%>
       <%= link_to "All Items", "/items"%>
-      <%= link_to "Cart: #{cart.total_items}", "/cart" %>
       <%= link_to "Register", "/users/new", method: :get %>
+      <% if current_admin? %>
+        <%= link_to "Admin Dashboard", "/admin" %>
+        <%= link_to "Users", "/admin/users" %>
+      <% elsif current_merchant? %>
+        <%= link_to "Merchant Dashboard", "/merchant/dashboard" %>
+      <% else %>
```
Comment (ryandbarnett) [inline app/views/layouts/application.html.erb:23]: Tabbed these over one.

File: config/routes.rb [hunk 4]
```diff
@@ -12,6 +12,12 @@
   patch "/merchants/:id", to: "merchants#update"
   delete "/merchants/:id", to: "merchants#destroy"
 
+  #this route is for US4 and the merchant dashboard
+#  get "/merchant", to: "dashboard#index"
+  namespace :merchant do
```
Comment (PhilipDeFraties) [inline config/routes.rb:17]: as per Tim I think line 18 should be changed to reflect the spec

File: config/routes.rb [hunk 5]
```diff
@@ -12,6 +12,11 @@
   patch "/merchants/:id", to: "merchants#update"
   delete "/merchants/:id", to: "merchants#destroy"
 
+  #this route is for US4 and the merchant dashboard
+  namespace :merchant do
+    get '/', to: 'dashboard#index'
+  end
```
Comment (PhilipDeFraties) [inline config/routes.rb:19]: Nice this makes sense!

**Review by ryandbarnett — COMMENTED** [review]
This was merged without the changes I asked for. Totally fine. I will make a small PR with the changes I think were needed.

**Pull request merged**